src/js/stores/MesosStateStore.js

```diff
@@ -194,18 +211,24 @@ class MesosStateStore extends GetSetBaseStore {
   }
 
   getTasksFromNodeID(nodeID) {
-    const services = this.get("lastMesosState").frameworks || [];
+    const frameworks = this.get("lastMesosState").frameworks || [];
     const memberTasks = {};
 
     const schedulerTasks = this.getSchedulerTasks();
 
-    services.forEach(service => {
-      service.tasks.forEach(function(task) {
+    frameworks.forEach(framework => {
+      framework.tasks.forEach(function(task) {
         if (task.slave_id === nodeID) {
-          memberTasks[task.id] = assignSchedulerTaskField(task, schedulerTasks);
+          // Need to get service from Marathon because we need the labels
+          const service = DCOSStore.serviceTree.findServiceByName(
```

Comment:
That just one more reason to introduce a custom find. Introducing a custom/specific find will also allow us to eventually remove the old "fundamentally broken" `findServiceByName` method.